packages/valory/skills/proposal_voter_abci/behaviours.py

```diff
@@ -216,41 +223,99 @@ class PrepareVoteTransactionBehaviour(ProposalVoterBaseBehaviour):
 
     matching_round: Type[AbstractRound] = PrepareVoteTransactionRound
 
+    def _get_proposal_info(self):
+        """Get the votable proposals' ids and the proposals."""
+        votable_proposal_ids = self.synchronized_data.votable_proposal_ids
+        proposals = self.synchronized_data.proposals
+
+        if self.synchronized_data.just_voted:
+            # Pending votes are stored in the shared state and only updated in the proposals list
+            # when the transaction has been verified, and therefore we know that it is a submitted vote.
+            submitted_vote = self.context.state.pending_vote
+            submitted_vote_id = submitted_vote.proposal_id
+            submitted_proposal = proposals[submitted_vote_id]
+            submitted_proposal["vote"] = submitted_vote.vote_choice
+            submitted_proposal["votable"] = submitted_vote.votable
```

Comment:
@dvilelaf would it be safe if we deleted them instead?

Reply:
I think we need to keep them for a while so the Fronted can show what the service voted on for a while, but I'll ask Oak